### DISCUSSION TITLE
Does jupedsim have the ability to simulate 3D space?

### DISCUSSION THREAD

**Discussion by verse-y** [PedestrianDynamics/jupedsim#1412] [General]
Can we set the Y-axis property for the pedestrian's position? By doing this, evacuation simulation across floors can be realized. If such a function is not available, drawing two geometric Spaces to represent two floors simultaneously is necessary, and pedestrians enter from one floor to another. How to achieve such an effect?

**Comment by Ozaq** [comment]
Right now JuPedSim is strictly 2D. All agents move in the x/y plane only. Further JuPedSim restricts the walkable area to be a single simple polygon with holes. 

With those limitations there is a workaround available, you can instantiate multiple Simulation objects, each with its own geometry, agents, stages and journeys. 

Agents can be moved from one floor to the next by removing them when they enter a space representing a staircase and adding them of the target floors simulation.
You could use an exit stage for egress points, in this case the simulation keeps track of agents that have been removed for you. You can get the ids of the agents that exited by calling `removed_agents()`. 

Note however that you will run into problems with this approach once the density in this hand over area is increasing.
